packages/connect/src/auth.ts

```diff
@@ -33,6 +33,24 @@ export interface AuthOptions {
   };
 }
 
+/**
+ * Firefox and mobile should not use a 'popup' type of window.
+ */
+export const shouldUsePopup = () => {
+  const isMobile = window.matchMedia('only screen and (max-width: 760px)').matches;
+  const isFirefox = navigator.userAgent.toLowerCase().includes('firefox');
+
+  return !isMobile && !isFirefox;
```

Comment:
You reckon it's worth using some (small) third party lib for this, or alternatively leaving it to the consumer to decide what is "mobile"?

Detecting mobile devices is notoriously difficult. Also, some users (especially developers with devtools open), will be < 760px. Which might be quite confusing.

Reply:
I've updated it - that media query didn't work the way I thought it did. I'm now using:

```js
!!('onTouchStart' in document.documentElement)
```

Which is, I believe, how `modernizr` works.

> or alternatively leaving it to the consumer to decide what is "mobile"?

We could - but I'm quite hesitant to ask devs to include the same code in every app, if we can just do it for them.

Reply:
Plenty of Windows desktops have touch displays, which'd be classed as mobiles with this check. https://stackoverflow.com/a/11381730/1141891 <-- think the most accurate way to determine mobile is to check for mobile browsers themselves.

Reply:
Aha, thanks, I hadn't considered that. I will change this (again) to use the UA check.